cloudmarker/stores/splunkhecstore.py

```diff
@@ -0,0 +1,111 @@
+"""SplunkStore plugin to index data in Splunk using HEC token."""
+
+import json
+import logging
+
+import requests
+
+_log = logging.getLogger(__name__)
+
+
+class SplunkHECStore:
+    """SplunkHECStore plugin to index cloud data in Splunk using HEC token."""
+
+    def __init__(self, uri, token, index_name, ca_cert, buffer_size=1000):
+        """Create an instance of :class:`SplunkHECStore` plugin.
+
+        Arguments:
+          uri (str): Splunk collector service URI.
+          token (str): Splunk HEC token.
+          index_name (str): Splunk HEC token accessible index.
+          ca_cert (str): Location of cetificate file to verify the identity
+            of host in URI, or Flase to disable verification
+          buffer_size (int): Maximum number of records to hold in
+            in-memory buffer for each record type.
+        """
+        self._splunk_uri = uri
+        self._splunk_token = token
+        self._splunk_index = index_name
+
+        self._ca_cert = ca_cert
+        self._buffer_size = buffer_size
+        self._buffer = []
+
+        # For maintaining session between multiple _flush calls
+        self._session = requests.session()
+
+    def write(self, record):
+        """Save the record in a bulk-buffer.
+
+        Also, flush the buffer by saving its content to Splunk when the buffer
+        size exceeds configured self._buffer_size
+
+        Arguments:
+          record (dict): Data to save to the Splunk.
+
+        """
+        # Make Splunk ready payload data and append it to self._buffers list.
+        self._buffer.append({
+            'index': self._splunk_index,
+            'sourcetype': 'json',
+            'event': record
+        })
+
+        # If the records count in self._buffer is more than allowed by
+        # self._buffer_size, send those records to Splunk.
+        if len(self._buffer) >= self._buffer_size:
+            self._flush()
+
+    def _flush(self):
+        """Perform bulk insert of buffered records into Splunk.
+
+        Arguments:
+          record_type (str): Record type (source name in Splunk).
+          records (list): List of records of type dict to index.
```

Comment:
Please remove the arguments list from the docstring. This method does not take any arguments anymore.